datastore/src/main/java/org/jboss/pnc/datastore/predicates/rsql/RSQLNodeTravellerPredicate.java

```diff
@@ -1,78 +1,91 @@
+import cz.jirutka.rsql.parser.ast.EqualNode;
+import cz.jirutka.rsql.parser.ast.InNode;
+import cz.jirutka.rsql.parser.ast.LogicalNode;
+import cz.jirutka.rsql.parser.ast.Node;
+import cz.jirutka.rsql.parser.ast.NotEqualNode;
+import cz.jirutka.rsql.parser.ast.NotInNode;
+import cz.jirutka.rsql.parser.ast.OrNode;
 
 public class RSQLNodeTravellerPredicate<Entity> implements RSQLPredicate {
 
-   private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
-   private final Node rootNode;
+    private final Node rootNode;
 
-   private final Class<Entity> selectingClass;
+    private final Class<Entity> selectingClass;
 
-   private final Map<Class<? extends ComparisonNode>, Transformer<Entity>> operations = new HashMap<>();
+    private final Map<Class<? extends ComparisonNode>, Transformer<Entity>> operations = new HashMap<>();
 
-   public RSQLNodeTravellerPredicate(Class<Entity> entityClass, String rsql) throws RSQLParserException {
-      operations.put(EqualNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).eq(arguments.get(0)));
-      operations.put(NotEqualNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).ne(arguments.get(0)));
-      operations.put(InNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).in(arguments));
-      operations.put(NotInNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).notIn(arguments));
+    public RSQLNodeTravellerPredicate(Class<Entity> entityClass, String rsql) throws RSQLParserException {
 
-      this.rootNode = new RSQLParser().parse(rsql);
-      this.selectingClass = entityClass;
-   }
+        operations.put(EqualNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).eq(arguments.get(0)));
+        operations.put(NotEqualNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).ne(arguments.get(0)));
+        operations.put(InNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).in(arguments));
+        operations.put(NotInNode.class, (pathBuilder, operand, arguments) -> pathBuilder.get(operand).notIn(arguments));
 
-   @Override
-   public BooleanExpression get() {
+        this.rootNode = new RSQLParser().parse(rsql);
+        this.selectingClass = entityClass;
+    }
 
-      PathBuilder<Entity> pathBuilder = new PathBuilder<>(selectingClass, selectingClass.getSimpleName().toLowerCase());
+    @Override
+    public BooleanExpression get() {
 
-      RSQLNodeTraveller<BooleanExpression> visitor = new RSQLNodeTraveller<BooleanExpression>() {
+        // Using lower-cases string variables makes Entities with camelCase names unusable (i.e. BuildRecord)
+        PathBuilder<Entity> pathBuilder = new PathBuilder<>(selectingClass, Introspector.decapitalize(selectingClass
+                .getSimpleName()));
 
```

Comment:
@slaskawi I had to change into Introspector.decapitalize(selectingClass.getSimpleName())), as the Entities with camelCase name could not be resolved. In this way I will put "buildRecord" and not "buildrecord", which does not get resolved